### PR TITLE
HTCONDOR-2373 keep file xfer plugin exit code in subcode

### PR DESCRIPTION
**Pull request by GregThain**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: docs/version-history/lts-versions-23-0.rst
Comment (Todd-L-Miller) [inline docs/version-history/lts-versions-23-0.rst:40]: Something like

```suggestion
- Fixed bug where :attr:`HoldReasonSubcode` was not its documented value for file transfer plugin errors.
  :jira:`2373`
```

would be less confusing, I think.

File: src/condor_tests/test_xfer_hold_codes.py
Comment (Todd-L-Miller) [inline src/condor_tests/test_xfer_hold_codes.py:126]: The current bug is that HoldReasonSubCode is _always_ `(1 << 8)`, so this test is not just mostly useless, but also deceptive.  You'll probably have to write a plugin that exits with different codes depending on its URLs to make sure the error reporting is working properly.

**Review by Todd-L-Miller — COMMENTED** [review]
It looks like
https://github.com/htcondor/htcondor/blob/V23_0-branch/src/condor_utils/file_transfer.cpp#L5454
isn't being modified by this PR, which is probably wrong.

There's another one missing, but it may be only on the main branch:
https://github.com/htcondor/htcondor/blob/main/src/condor_utils/file_transfer.cpp#L5606

File: src/condor_tests/test_xfer_hold_codes.py
Comment (GregThain) [reply src/condor_tests/test_xfer_hold_codes.py:126]: That test is coming on another ticket, but this was a quick and dirty way to make sure we don't completely regress the SubCode code.

File: docs/version-history/lts-versions-23-0.rst
Comment (GregThain) [reply docs/version-history/lts-versions-23-0.rst:40]: Note we don't have the :attr: tags in the stable branch of the docs.

**Review by Todd-L-Miller — APPROVED** [review]
Looks good.

It's probably not worth the trouble to change the release note in the dev-series manual, but there should probably be another ticket to address the error message that will become incorrect after this merges to the main branch.

The regression test being completely broken definitely needs its own ticket.  (Checking to see that the hold subcode is nonzero would be more honest and less wrong, but still not a real test.)

Approved pending a successful BaTLab run.

**Pull request merged**
